crates/utils/re_mcap/src/parsers/ros2msg/sensor_msgs/compressed_image.rs

```diff
@@ -12,19 +17,96 @@ use crate::parsers::{
 };
 use crate::util::TimestampCell;
 
+const CONFIG_HEADER_SIZE: usize = size_of::<i32>() + size_of::<[f32; 2]>();
+const RESOLUTION_HEADER_SIZE: usize = size_of::<[u32; 2]>();
+
+/// Metadata extracted from a ROS2 `compressedDepth` RVL payload.
+#[derive(Clone, Copy, Debug, PartialEq)]
+struct RvlMetadata {
+    pub width: u32,
+    pub height: u32,
+    pub depth_quant_a: f32,
+    pub depth_quant_b: f32,
+    payload_offset: usize,
+    num_pixels: usize,
+}
+
+#[derive(Debug, thiserror::Error)]
+enum RvlDecodeError {
+    #[error("compressed depth payload missing RVL header")]
+    MissingHeader,
+
+    #[error("RVL payload missing resolution header")]
+    MissingResolution,
+
+    #[error("RVL payload reports zero resolution")]
+    ZeroResolution,
+
+    #[error("RVL image resolution would overflow")]
+    ResolutionOverflow,
+
+    #[error("RVL payload shorter than expected for resolution {width}x{height}")]
+    PayloadLengthMismatch { width: u32, height: u32 },
+}
+
+fn parse_ros_rvl_metadata(data: &[u8]) -> Result<RvlMetadata, RvlDecodeError> {
```

Comment:
IIUC this is the same metadata parsing from `crates/viewer/re_viewer_context/src/utils/depth.rs` right? Is there any way we can avoid duplicating this?